telemetry/protocolstats.go

```diff
@@ -0,0 +1,56 @@
+package telemetry
+
+import (
+	"database/sql"
+	"time"
+)
+
+type Metric struct {
+	TotalIn  int64   `json:"totalIn"`
+	TotalOut int64   `json:"totalOut"`
+	RateIn   float64 `json:"rateIn"`
+	RateOut  float64 `json:"rateOut"`
+}
+
+type ProtocolStats struct {
+	HostID string `json:"hostID"`
+	Relay  Metric `json:"relay"`
+	Store  Metric `json:"store"`
+}
+
+func (r *ProtocolStats) insertRate(db *sql.DB, protocolName string, metric Metric) error {
+	stmt, err := db.Prepare("INSERT INTO protocolStatsRate (hostId, protocolName, rateIn, rateOut, createdAt) VALUES ($1, $2, $3, $4, $5);")
+	if err != nil {
+		return err
+	}
+	_, err = stmt.Exec(r.HostID, protocolName, metric.RateIn, metric.RateOut, time.Now().Unix())
+	if err != nil {
+		return err
+	}
+
+	stmt, err = db.Prepare("INSERT INTO protocolStatsTotals (hostId, protocolName, totalIn, totalOut, createdAt) VALUES ($1, $2, $3, $4, $5) ON CONFLICT ON CONSTRAINT protocolStatsTotals_unique DO UPDATE SET totalIn = $3, totalOut = $4;")
```

Comment:
it would be total overall? don't u want to add this over a period of time?

Reply:
hm. good point. I'll change it so it's total per day

Reply:
I noticed that totalIn / totalOut variables in libp2p are accumulated values, so in https://github.com/status-im/status-go/pull/3153/commits/0e8a9f11bbfb87ba2f33841b49e6a8a654b5efb1 i'm resetting the totals every day. This will naturally agregate the data per user per day in protocolStatsTotal, and later with some sql wizardry we can get proper reports